applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.cpp

```diff
@@ -593,18 +596,27 @@ double AdjointFiniteDifferencingBaseElement::GetPerturbationSizeModificationFact
     KRATOS_CATCH("")
 }
 
-void AdjointFiniteDifferencingBaseElement::save(Serializer& rSerializer) const
+template <class TPrimalElement>
+void AdjointFiniteDifferencingBaseElement<TPrimalElement>::save(Serializer& rSerializer) const
 {
     KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer,  Element );
     rSerializer.save("mpPrimalElement", mpPrimalElement);
+    rSerializer.save("mHasRotationDofs", mHasRotationDofs);
 }
 
-void AdjointFiniteDifferencingBaseElement::load(Serializer& rSerializer)
+template <class TPrimalElement>
+void AdjointFiniteDifferencingBaseElement<TPrimalElement>::load(Serializer& rSerializer)
 {
     KRATOS_SERIALIZE_LOAD_BASE_CLASS( rSerializer, Element );
     rSerializer.load("mpPrimalElement", mpPrimalElement);
+    rSerializer.load("mHasRotationDofs", mHasRotationDofs);
 
 }
 
+template class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) AdjointFiniteDifferencingBaseElement<ShellThinElement3D3N>;
```

Comment:
@msandre Here i define all the specializations with the different primal element types that are used by the derived classes and therefor use these base class methods.

Reply:
If this at some point blows up then you could make this class header-only

Reply:
:+1: